src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -112,34 +115,110 @@ private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
                 dotnetPath = Path.GetDirectoryName(isWindows ? dotnetPath : realpath(dotnetPath) ?? dotnetPath);
             }
 
+            string pathOfPinnedSdk = string.Empty;
+
+            foreach (string path in GetDotNetBasePaths(workingDirectory, dotnetPath))
```

Comment:
What's your thinking on preferring the PATH search to the current executable? I could see that going either way.

Reply:
I agree it could go either way...in this case, since I'm adding them all to a HashSet rather than a List (for deduplication), I'm not actually preferring one above the other. I'm just going by version number and mixing them up. (I also see an argument for keeping them separate...but I'm not sure how valuable that is, and I honestly don't think the "from current process" code path will be important all that often.)